components/Users/Memberships/index.js

```diff
@@ -120,7 +122,68 @@ const getState = (membership) => {
   return 'aktiv'
 }
 
+
+const SET_AUTO_PAY = gql`
+  mutation setMembershipAutoPay($id: ID!, $autoPay: Boolean!) {
+    setMembershipAutoPay(id: $id, autoPay: $autoPay) {
+      id
+      autoPay
+    }
+  }
+`
+
+function AutoPayToggle (membership) {
+
+  if (!membership.autoPayIsMutable) return (
+    <React.Fragment>
+      <DT>
+        Automatisch abbuchen
+      </DT>
+      <DD>
+        { membership.autoPay ? 'YES' : 'NO' }
+      </DD>
+    </React.Fragment>
+  )
+
+  return (
+    <Mutation
+      mutation={SET_AUTO_PAY}
+      variables={{ id: membership.id, autoPay: !membership.autoPay }}
```

Comment:
It would be better to listen to which option was actually selected, therefore I'd remove the variables here:
```suggestion
```
and then see the next suggestion in the `Dropdown.onChange` handler.

Reply:
I struggle to see the difference it makes. Could you elaborate?

Reply:
To avoid race conditions it's better to listen to what the user selected instead of assuming that they selected the opposite of the server state.